exdir/core/constants.py

```diff
@@ -2,6 +2,15 @@
 EXDIR_METANAME = "exdir"
 TYPE_METANAME = "type"
 VERSION_METANAME = "version"
+LINK_METANAME = "link"
```

Comment:
I think I prefer if the type is `soft_link`, `external_link`, etc. That reduces the level of conditional nesting - you can say `if type == "external_link"` instead of `if type == "link" and link_type == "external"`. And then have the meta-groups be `soft_link` and `external_link` as well. It just reduces the confusion of which variables can go together. (No-one will think that they can just change the target from `external` to `soft` and still keep the `file`, for instance.